density/db/db.py

```diff
@@ -210,3 +211,39 @@ def get_uni_for_code(cursor, code):
     result = cursor.fetchone()
     if result is not None:
         return result['uni']
+
+PARENTS = {
+  '79' : 'Lehman Library',
+  '84' : 'Lerner',
+  '15' : 'Northwest Corner Building',
+  '75' : 'John Jay',
+  '103' : 'Butler',
+  '131' : '',
+  '146' : 'Avery',
+  '62' : 'East Asian Library',
+  '2' : 'Uris'
+}
+
+def insert_density_data(cursor, data):
+  query = """INSERT INTO {table_name}
+             (dump_time, group_id, group_name, parent_id, parent_name, client_count)
+             VALUES %s;""".format(table_name=TABLE_NAME)
+
+  # Set the time
+  date = datetime.now().replace(second=0, microsecond=0)
+
+  db_failed = 0
+
+  # NOTE: We tried batching these but ran into some problems escaping quotes.
+  # It's easy with list comprehensions to batch them, but hard to escape
+  # single quotes that may appear in group names.
+  for key in data:
+    query = """INSERT INTO {table_name}
+             (dump_time, group_id, group_name, parent_id, parent_name, client_count)
+             VALUES (%s, %s, %s, %s, %s, %s);""".format(table_name=TABLE_NAME)
+    try:
+      cursor.execute(query, (date, int(key), data[key]['name'], int(data[key]['parent_id']), PARENTS[data[key]['parent_id']], int(data[key]['client_count'])))
+    except:
+      db_failed += 1
```

Comment:
Why are we swallowing a possible failed insertion error? why not just bubble it up and fail the request if necessary?

If we are going to swallow the error, we should at least log it for our own sanity.